pkgs/by-name/on/onedrive/package.nix

```diff
@@ -17,7 +18,7 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "onedrive";
-  version = "2.5.3";
+  version = "2.5.3-1";
```

Comment:
Apologies. Does the version not need to be incremented somehow to ensure that people who already have 2.5.3 installed receive the patch? Or does nixpkgs handle that itself?

Reply:
> Apologies. Does the version not need to be incremented somehow to ensure that people who already have 2.5.3 installed receive the patch? Or does nixpkgs handle that itself?

Well, no, all packages are delivered to users as a whole, and switching between two Nixpkgs checkouts is atomic.

But I just suddenly realized that your method might be useful for tracking security updates, if implemented correctly.

Reply:
Okay yeh that makes sense. Will I just leave as 2.5.3 anyway? Sounds like a larger discussion for nixpkgs to implement -x versions

Reply:
Our version generally refers to the upstream version. If you discuss this issue with me here, there will probably be no results.

Reply:
okay thanks. I've left it at 2.5.3 for now.

Reply:
> Does the version not need to be incremented somehow to ensure that people who already have 2.5.3 installed receive the patch?

I think it would be worse in the end.

Patches and other source modifications are treated as phases, they are part of the build script.
Mangling the "downstream patch count" (DPC) into `version` string would not be useful for us, since we use `version` mainly to parameterize the pristine `src`.

On the other hand we could treat this DPC as an extra attribute.
However, the same file can be modified multiple times by different people (merge conflicts and deadlocks, anyone?).

The best I can think on this ground is a global counter, extracted from Git metadata.
But this becomes redundant - to update this global counter, we should  update the Git repo itself.

> Or does nixpkgs handle that itself?

Yes: when the user updates the channels and run the `nixos-rebuild` commands, they will grab the updated channels.